src/mpid/ch3/channels/nemesis/src/mpid_nem_init.c

```diff
@@ -238,24 +241,42 @@ MPID_nem_init(int pg_rank, MPIDI_PG_t *pg_p, int has_parent ATTRIBUTE((unused)))
 #endif  /*FORCE_ASYM */
 
     /* Request fastboxes region */
-    mpi_errno = MPIDU_shm_seg_alloc(MPL_MAX((num_local*((num_local-1)*sizeof(MPID_nem_fastbox_t))), MPID_NEM_ASYMM_NULL_VAL),
-                                     (void **)&fastboxes_p, MPL_MEM_SHM);
+#ifdef HAVE_HWLOC
+    MPIR_Info_get_impl(MPIR_Process.obj_mbind, "fastboxes", MPI_MAX_INFO_VAL, info_value, &info_flag);
+    if (info_flag)
+        kind = (strcmp(info_value, "mcdram") == 0 && MPIR_Process.memkind_is_avail[MPIR_MEMKIND__MCDRAM]) ?
+            MPIR_MEMKIND__MCDRAM : MPIR_MEMKIND__DRAM;
```

Comment:
Can you add an error case if the user picks an unsupported value, e.g., `foobar`?  Also, if the user picked `mcdram` and there's no `MCDRAM` available, you are silently falling back to `DRAM`.  Is that the right thing to do or should we throw an error?  I'm not sure what the answer is.  Maybe others in the group have an opinion.

Btw, on what is this info set?  Since fastbox/cells are not specific to a communicator or a window, I'm not sure info keys make too much sense here.  I'm sorry if I was the one who suggested doing that in the first place.  It sounds like that was bad advice.  :-)

Reply:
I am not familiar with the error handling inside MPICH can you provide some hint on how this should be done. I suspect I need to create an error code and then pass it to upper layers with `MPIR_ERR_POP`?

Reply:
We should probably discuss what object this info is set on first.  Because, if that's the wrong thing to do, you can probably entirely delete this part of the code.

Reply:
Sure